diff -rupN gcc-4.4.3-orig/config.sub gcc-4.4.3/config.sub
--- gcc-4.4.3-orig/config.sub	2010-02-04 10:01:17.000000000 +0100
+++ gcc-4.4.3/config.sub	2010-02-04 10:03:32.836181575 +0100
@@ -1278,7 +1278,7 @@ case $os in
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -solaris* | -sym* \
-	      | -kopensolaris* \
+	      | -kopensolaris* | -tylyos* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
 	      | -aos* | -aros* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
diff -rupN gcc-4.4.3-orig/gcc/config/tylyos.h gcc-4.4.3/gcc/config/tylyos.h
--- gcc-4.4.3-orig/gcc/config/tylyos.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.4.3/gcc/config/tylyos.h	2010-02-04 10:39:40.000000000 +0100
@@ -0,0 +1,11 @@
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()      \
+  do {                                \
+    builtin_define_std ("tylyos");      \
+    builtin_define_std ("unix");      \
+    builtin_assert ("system=tylyos");   \
+    builtin_assert ("system=unix");   \
+  } while(0);
+
+#undef TARGET_VERSION
+#define TARGET_VERSION fprintf(stderr, " (i386 tylyos)");
diff -rupN gcc-4.4.3-orig/gcc/config.gcc gcc-4.4.3/gcc/config.gcc
--- gcc-4.4.3-orig/gcc/config.gcc	2010-02-04 10:01:15.000000000 +0100
+++ gcc-4.4.3/gcc/config.gcc	2010-02-04 10:38:17.083604078 +0100
@@ -407,6 +407,12 @@ esac
 
 # Common parts for widely ported systems.
 case ${target} in
+*-*-tylyos*)
+  extra_parts="crtbegin.o crtend.o"
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  ;;
 *-*-darwin*)
   tm_file="${tm_file} darwin.h"
   case ${target} in
@@ -602,6 +608,11 @@ case ${target} in
 	then tmake_file=${cpu_type}/t-$rest
 	fi
 	;;
+i[3-7]86-*-tylyos*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h tylyos.h"
+	tmake_file="i386/t-i386elf t-svr4"
+	use_fixproto=yes
+	;;
 alpha*-*-linux*)
 	tm_file="${tm_file} alpha/elf.h alpha/linux.h alpha/linux-elf.h"
 	target_cpu_default="MASK_GAS"
diff -rupN gcc-4.4.3-orig/libgcc/config.host gcc-4.4.3/libgcc/config.host
--- gcc-4.4.3-orig/libgcc/config.host	2010-02-04 10:01:21.000000000 +0100
+++ gcc-4.4.3/libgcc/config.host	2010-02-04 10:46:41.547180523 +0100
@@ -172,6 +172,8 @@ case ${host} in
 	then tmake_file=${cpu_type}/t-$rest
 	fi
 	;;
+i[3-7]86-*-tylyos*)
+	;;
 alpha*-*-linux* | alpha*-*-gnu*)
 	tmake_file="${tmake_file} alpha/t-crtfm"
 	extra_parts="$extra_parts crtfastmath.o"
